### PR TITLE
DOC: update typos and grammar errors

### PR DESCRIPTION
**Pull request by AdvancedUno**
## PR Summary

#24762 
Fix a typo and grammar errors on #24762

```diff
- Making an new branch for each set of related changes will make it easier for
+ Making a new branch for each set of related changes will make it easier for

-Enter a title for the set of changes, and some explanation of what you've done.
-Say if there is anything you'd like particular attention for - like a
+Enter a title for a set of changes with some explanation of what you've done.
+Mention anything you'd like particular attention for - such as a

-There's an embarrassing typo in a commit you made? Or perhaps the you
-made several false starts you would like the posterity not to see.
+Is there any embarrassing typo in a commit you made? Or perhaps you
+made several false starts you don't want posterity to see.

-and the history looks now like this::
+and now, history looks like this::

-history will get messy. Most of the time you can just ask for a review, and not
-worry that ``main`` has got a little ahead.  But sometimes, the changes in
+history will get messy. Most of the time, you can just ask for a review without
+worrying about whether ``main`` has got a little ahead; however sometimes, the changes in

-When all looks good you can delete your backup branch::
+When all looks good, you can delete your backup branch::

-branch *cool-feature*.  If you try to push the new commits to GitHub it will
-fail with an error that looks like ::
+branch *cool-feature*.  If you try to push the new commits to GitHub, it will
+fail and show an error that looks like ::

-If this push had succeed then the commits ``A``, ``B``, and ``C`` would no
-longer be referenced by any branch and be discarded:
+If this push had succeeded, the commits ``A``, ``B``, and ``C`` would no
+longer be referenced by any branch and they would be discarded:

-However, in the case of having intentionally re-written history we *want* to
+However, in the case of having intentionally re-written history, we *want* to

-versions from our local branch.  In this case, what we want to do is ::
+versions from our local branches.  In this case, what we want to do is ::

-history and if anyone has fetched the old commits will have a different view
+history, and if anyone has fetched the old commits, it will have a different view
```

## PR Checklist

**Documentation and Tests**
- [N/A] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: doc/devel/development_workflow.rst [hunk 1]
```diff
@@ -415,9 +415,9 @@ GitHub also adds the helpful suggestion to pull the remote changes and then try
 pushing again.  In some cases, such as if you and a colleague are both
 committing and pushing to the same branch, this is a correct course of action.
 
-However, in the case of having intentionally re-written history we *want* to
+However, in the case of having intentionally re-written history, we *want* to
 discard the commits on the remote and replace them with the new-and-improved
-versions from our local branch.  In this case, what we want to do is ::
+versions from our local branches.  In this case, what we want to do is ::
```
Comment (QuLogic) [inline doc/devel/development_workflow.rst:420]: We're only talking about a single branch.
```suggestion
versions from our local branch.  In this case, what we want to do is ::
```

Comment (AdvancedUno) [reply doc/devel/development_workflow.rst:420]: Thank you for your help!

**Comment by AdvancedUno** [comment]
@QuLogic  Thank you for your help!

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @AdvancedUno and congratulations on your first contribution to Matplotlib!

**Comment by AdvancedUno** [comment]
@timhoffm Thank you!! Glad to be helpful!

**Comment by saranti** [comment]
@QuLogic 

> I'm going to assume this is fully fixed by #24782.

There are still some issues:
> ```diff
>-Enter a title for a set of changes with some explanation of what you've done.
>-Mention anything you'd like particular attention for - such as a
>+Enter a title for the set of changes with some explanation of what you've done.
>+Mention anything you'd like particular attention paid to - such as a
>
>-worrying about whether ``main`` has got a little ahead; however sometimes, the changes in
>-``main`` might affect your changes, and you need to harmonize them.  In this
>-situation you may prefer to do a rebase.
>+worrying about whether ``main`` has gotten a little ahead; however sometimes, the changes in
>+``main`` might affect your changes and you need to harmonize them.  In this
>+situation, you may prefer to do a rebase.
>
>-fail and show an error that looks like ::
>+fail and show an error that looks like::
>
>-versions from our local branch.  In this case, what we want to do is ::
>+versions from our local branch.  In this case, what we want to do is::
>
>-history, and if anyone has fetched the old commits, it will have a different view
>+history, and if anyone has fetched the old commits, they will have a different view

After this, it'll be perfect

**Pull request merged**